src/backend/updates.rs

```diff
@@ -1322,3 +673,19 @@ fn check_candid_interface_compatibility() {
         candid_parser::utils::CandidSource::File(old_interface.as_path()),
     );
 }
+
+#[export_name = "canister_update archive"]
+fn archive() {
+    mutate(|state| {
+        if !state.principal_to_user(caller()).unwrap().stalwart {
+            return;
+        }
+        let n: usize = parse(&arg_data_raw());
```

Comment:
Even if a stalwart is malicious, then cannot do any harm here, right? Too small n and too large n both work properly.